src/jit/compiler.cpp

```diff
@@ -516,64 +487,450 @@ var_types    Compiler::argOrReturnTypeForStruct(unsigned size, CORINFO_CLASS_HAN
+                *wbPassStruct = SPK_ByValue;
+                useType = TYP_UNKNOWN;
+
+#else  //  _TARGET_XXX_
+
+                noway_assert(!"Unhandled TARGET in getArgTypeForStruct (with FEATURE_MULTIREG_ARGS=1)");
+
+#endif  //  _TARGET_XXX_
+
             }
         }
-#endif // FEATURE_MULTIREG_RET
+        else  // (structSize > MAX_PASS_MULTIREG_BYTES)
+        {
+            // We have a (large) struct that can't be replaced with a "primitive" type
+            // and can't be passed in multiple registers
 
-#if FEATURE_MULTIREG_ARGS
-        if (!forReturn)
+#if defined(_TARGET_X86_) || defined(_TARGET_ARM_)
+
+            // Otherwise we pass this struct by value on the stack 
+            // setup wbPassType and useType indicate that this is passed by value according to the X86/ARM32 ABI
+            *wbPassStruct = SPK_ByValue;
+            useType = TYP_UNKNOWN;
+
+#elif defined(_TARGET_AMD64_) || defined(_TARGET_ARM64_)
+
+            // Otherwise we pass this struct by reference to a copy
+            // setup wbPassType and useType indicate that this is passed using one register (by reference to a copy)
+            *wbPassStruct = SPK_ByReference;
+            useType = TYP_UNKNOWN;
+
+#else  //  _TARGET_XXX_
+
+            noway_assert(!"Unhandled TARGET in getArgTypeForStruct");
+
+#endif  //  _TARGET_XXX_
+
+        }
+    }
+
+    assert(*wbPassStruct != SPK_Unknown);  // This must get set to one of the valid values before we return
+    return useType;
+}
+
+//-----------------------------------------------------------------------------
+// getReturnTypeForStruct: 
+//     Get the type that is used to return values of the given struct type.
+//     If you already have retrieved the struct size then it should be passed
+//     as the third optional argument as it allows us to avoid an extra call
+//     to getClassSize(clsHnd)
+//
+// Arguments:
+//    clsHnd         - the handle for the struct type
+//    wbReturnStruct - information about how the struct is returned
+//    structSize     - the size of the struct type, 
+//                     or 0 if we should to call getClassSize(clsHnd)
+//
+// Return Value:
+//    When *wbReturnStruct is SPK_PrimitiveType this method's return value:
+//       is the primitive type used to return the struct.
+//    When *wbReturnStruct is SPK_ByReference this method's return value:
+//       is TYP_UNKNOWN when the struct type is returned using a return buffer
+//    When *wbReturnStruct is SPK_ByValue or SPK_ByValueAsHfa the return value:
+//       is TYP_STRUCT when the struct type is returned using multiple registers.
+//
+// Assumptions:
+//    The size must be the size of the given type.
+//    The given class handle must be for a value type (struct).
+//
+// Notes:
+//    About returning TYP_STRUCT:
+//        Whenever the return type is TYP_STRUCT it always means multiple registers
```

Comment:
"this method's return value" means the value returned by  getArgTypeForStruct()
The only occurances of "return type" remainig in my comments is this phrase:
"and the target supports multireg return types then the return value is TYP_STRUCT"


The comments that you refered to now reads:

// Return Value:
//    For wbPassStruct you can pass a 'nullptr' and nothing will be written
//     or returned for that out parameter.
//    When *wbPassStruct is SPK_PrimitiveType this method's return value
//       is the primitive type used to pass the struct.
//    When *wbPassStruct is SPK_ByReference this method's return value
//       is TYP_UNKNOWN and the struct type is passed by reference to a copy
//    When *wbPassStruct is SPK_ByValue or SPK_ByValueAsHfa this method's return value 
//       can be TYP_STRUCT and the struct type is passed using multiple registers.
//       or can be TYP_UNKNOWN and the struct type is passed by value (for x86 and ARM32)